src/lib/pickles_types/opt.ml

```diff
@@ -70,6 +70,9 @@ let map t ~f =
   | Maybe (b, x) ->
       Maybe (b, f x)
 
+let iter t ~f =
```

Comment:
Nit: it might be more helpful in future to have `f : 'bool Option.t -> 'a -> unit`, but LGTM for now

Reply:
Opened follow-up issue: https://github.com/MinaProtocol/mina/issues/14092